Add Getting Started documentation (#17)
sailfishos-chum/main: GettingStarted.md

# SailfishOS Chum - Getting Started with OBS

This document summarises the proces of building your first package on 
OBS.  It only skims the surface of OBS and is not a manual.

To start building your packages on OBS you will require an account.
Follow the process in the README and contact 'lbt' for an account to 
be set up.

When your account is set up, you will have a "Home Project".  This is where
you can start creating packages.  The URL for your home project will be:

> https://build.merproject.org/project/show/home:<username>

Reviewer: ```suggestion
https://build.sailfishos.org/project/show/home:<username>
```
Would it be possible to use `https://build.sailfishos.org/project/show/home:` instead? This is now the preferred domain.

## Creating a package on OBS

> Projects contain packages which are built against target repositories

Reviewer 2: Consider clarifying "package/rpm repositories"... i keep thinking git.

(sailfish versions).  To create your first package, perform the following
steps:

1. Click "Create Package"
2. On the new package page, give the package a name and click "Save Changes"
   * eg harbour-mypackage<br>

> You now need to add files to the package to allow OBS to get the source and

Reviewer: To get around markdown's list formatting restrictions, maybe this line could be added to 3 instead (the <br> doesn't seem to leave the correct indent)?

build it.

> 3. Create a file on your computer called _service with th e following content:

Reviewer: ```suggestion
3. Create a file on your computer called `_service` with the following content:
```
Typo and formatting.

>         <services>
>           <service name="tar_git">
>             <param name="url">URL TO PROJECT</param>
>             <param name="branch">master</param>
>             <param name="revision"></param>
>             <param name="token"/>
>             <param name="debian">N</param>
>             <param name="dumb">N</param>
>           </service>
>         </services>`

Reviewer: ````suggestion
```
        <services>
          <service name="tar_git">
            <param name="url">URL TO PROJECT</param>
            <param name="branch">master</param>
            <param name="revision"></param>
            <param name="token"/>
            <param name="debian">N</param>
            <param name="dumb">N</param>
          </service>
        </services>`
```
````
These lines need surrounding in "code block" formatters. Perhaps it would also help to clarify what `N` is.

Author: Probably those last 3 params are not even needed

  And specify at least the URL.  If you want to build another branch, change the 
  branch name, and if you want to build a particular revision, specify a git hash
  or tag in the revision.

> 4. Back on OBS, click "Add File" then "Browse" and select the _service file you

Reviewer: ```suggestion
4. Back on OBS, click "Add File" then "Browse" and select the `_service` file you
```
Possible format change. Not sure if this is really correct, but I think it's helpful personally.

   just created and "Save Changes"
5. If all goes well, OBS will fetch the source code.

## Adding target repositories

Back at the home project on OBS, you will require target repositories to build against.
These can be added using a UI but it is easier to add them manually.

1. Click "Advanced" and "Meta"<br>

>    Note, dont confuse this with the package Meta configuration, the nanes are the same

Reviewer: ```suggestion
   Note, dont confuse this with the package Meta configuration, the nanes are the same.
```

2. You will be presented with an XML description of the target repositories.  Between

>    the <project> </project> tags, add the following:
>         <repository name="sailfish_latest_i486">
>             <path project="sailfishos:latest" repository="latest_i486"/>
>             <arch>i586</arch>
>         </repository>
>         <repository name="sailfish_latest_armv7hl">
>             <path project="sailfishos:latest" repository="latest_armv7hl"/>
>             <arch>armv8el</arch>
>         </repository>
>         <repository name="sailfish_latest_aarch64">
>             <path project="sailfishos:latest" repository="latest_aarch64"/>
>             <arch>aarch64</arch>
>         </repository>
>         <repository name="sailfish_4.1.0.24_armv7hl">
>             <path project="sailfishos:4.1.0.24" repository="latest_armv7hl"/>
>             <arch>armv8el</arch>
>         </repository>
>         <repository name="sailfish_4.1.0.24_aarch64">
>             <path project="sailfishos:4.1.0.24" repository="latest_aarch64"/>
>             <arch>aarch64</arch>
>         </repository>
>         <repository name="sailfish_4.1.0.24_i486">
>             <path project="sailfishos:4.1.0.24" repository="latest_i408"/>
>             <arch>i586</arch>
>         </repository>`

Reviewer: Some more code block formatting.
````suggestion
      the `<project> </project>` tags, add the following:
```
        <repository name="sailfish_latest_i486">
            <path project="sailfishos:latest" repository="latest_i486"/>
            <arch>i586</arch>
        </repository>
        <repository name="sailfish_latest_armv7hl">
            <path project="sailfishos:latest" repository="latest_armv7hl"/>
            <arch>armv8el</arch>
        </repository>
        <repository name="sailfish_latest_aarch64">
            <path project="sailfishos:latest" repository="latest_aarch64"/>
            <arch>aarch64</arch>
        </repository>
        <repository name="sailfish_4.1.0.24_armv7hl">
            <path project="sailfishos:4.1.0.24" repository="latest_armv7hl"/>
            <arch>armv8el</arch>
        </repository>
        <repository name="sailfish_4.1.0.24_aarch64">
            <path project="sailfishos:4.1.0.24" repository="latest_aarch64"/>
            <arch>aarch64</arch>
        </repository>
        <repository name="sailfish_4.1.0.24_i486">
            <path project="sailfishos:4.1.0.24" repository="latest_i408"/>
            <arch>i586</arch>
        </repository>`
```
````

   This will create build targets for the 4.1 and latest SaifishOS releases for

>    all supported archutectures.

Reviewer: ```suggestion
   all supported architectures.
```

3. Click "Save Changes", navigate back to the project overview, and you should see
   the build status for each target on the right hand side.
4. You can click on the package name, and then click on the build target status to
   see the current build log.
  
## Submitting your package to saailfishos:chum:testing

> Once you are happyy your builds are working, you can submit it for inclusion to chum:testing.

Reviewer: ```suggestion
Once you are happy your builds are working, you can submit it for inclusion to chum:testing.
```

> 1. Naviate to the package, and click "Submit Package"

Reviewer: ```suggestion
1. Navigate to the package, and click "Submit Package"
```

2. Enter the target project as `sailfishos:chum:testing`
3. Click Ok

For packages submitted to chum, it is probably best to specify a particular git revision to be built.
For this reason, you may want to set up your home project with builds for your master
branch for personal use, and a specific revision to be submitted to chum.

## Other useful notes

* You can create sub-projects in your home project to group builds together
* You can add the repository for the builds to your device using `zypper ar` to install
  on your device
* To re-trigger OBS to download the latest code, click the "Trigger Service" link in the package
* You can configure OBS to not build certain packages against certain repositories, or disable

>   certain projects from being published.  It is very flexible.  This is available from the "Repsitories"

Reviewer: ```suggestion
  certain projects from being published.  It is very flexible.  This is available from the "Repositories"
```

>   link.

Reviewer: Would it be worth adding something about dependencies (i.e. if I want my project built against libraries in the chum repos)?

# SailfishOS Chum - Getting Started with OBS

This document summarises the proces of building your first package on 
OBS.  It only skims the surface of OBS and is not a manual.

To start building your packages on OBS you will require an account.
Follow the process in the README and contact 'lbt' for an account to 
be set up.

When your account is set up, you will have a "Home Project".  This is where
you can start creating packages.  The URL for your home project will be:
https://build.sailfishos.org/project/show/home:<username>

## Creating a package on OBS

Projects contain packages which are built against target repositories
(sailfish versions).  To create your first package, perform the following
steps:

1. Click "Create Package"
2. On the new package page, give the package a name and click "Save Changes"
   * eg harbour-mypackage
3. You now need to add files to the package to allow OBS to get the source and
build it.  Create a file on your computer called `_service` with the following content:

> ```
> <services>
>     <service name="tar_git">
>     <param name="url">URL TO PROJECT</param>
>     <param name="branch">master</param>
>     <param name="revision"></param>
>     </service>
> </services>`
> ```

Reviewer: See my other comment, the same applies here.
````suggestion
    ```
    <services>
        <service name="tar_git">
        <param name="url">URL TO PROJECT</param>
        <param name="branch">master</param>
        <param name="revision"></param>
        </service>
    </services>`
    ```
````

   And specify at least the URL.  If you want to build another branch, change the 
   branch name, and if you want to build a particular revision, specify a git hash
   or tag in the revision.
4. Back on OBS, click "Add File" then "Browse" and select the `_service` file you
   just created and "Save Changes"
5. If all goes well, OBS will fetch the source code.

## Adding target repositories

Back at the home project on OBS, you will require target repositories to build against.
These can be added using a UI but it is easier to add them manually.

1. Click "Advanced" and "Meta"<br>
   Note, dont confuse this with the package Meta configuration, the nanes are the same
2. You will be presented with an XML description of the target repositories.  Between
   the ```<project> </project>``` tags, add the following:

> ```
> <repository name="sailfish_latest_i486">
>     <path project="sailfishos:latest" repository="latest_i486"/>
>     <arch>i586</arch>
> </repository>
> <repository name="sailfish_latest_armv7hl">
>     <path project="sailfishos:latest" repository="latest_armv7hl"/>
>     <arch>armv8el</arch>
> </repository>
> <repository name="sailfish_latest_aarch64">
>     <path project="sailfishos:latest" repository="latest_aarch64"/>
>     <arch>aarch64</arch>
> </repository>
> <repository name="sailfish_4.1.0.24_armv7hl">
>     <path project="sailfishos:4.1.0.24" repository="latest_armv7hl"/>
>     <arch>armv8el</arch>
> </repository>
> <repository name="sailfish_4.1.0.24_aarch64">
>     <path project="sailfishos:4.1.0.24" repository="latest_aarch64"/>
>     <arch>aarch64</arch>
> </repository>
> <repository name="sailfish_4.1.0.24_i486">
>     <path project="sailfishos:4.1.0.24" repository="latest_i408"/>
>     <arch>i586</arch>
> </repository>
> ```

Reviewer: The proceeding list items get messed up because of the code block. It seems [the solution](https://gist.github.com/clintel/1155906) is to indent the fenced code block (suggestion untested).

````suggestion
    ```
    <repository name="sailfish_latest_i486">
        <path project="sailfishos:latest" repository="latest_i486"/>
        <arch>i586</arch>
    </repository>
    <repository name="sailfish_latest_armv7hl">
        <path project="sailfishos:latest" repository="latest_armv7hl"/>
        <arch>armv8el</arch>
    </repository>
    <repository name="sailfish_latest_aarch64">
        <path project="sailfishos:latest" repository="latest_aarch64"/>
        <arch>aarch64</arch>
    </repository>
    <repository name="sailfish_4.1.0.24_armv7hl">
        <path project="sailfishos:4.1.0.24" repository="latest_armv7hl"/>
        <arch>armv8el</arch>
    </repository>
    <repository name="sailfish_4.1.0.24_aarch64">
        <path project="sailfishos:4.1.0.24" repository="latest_aarch64"/>
        <arch>aarch64</arch>
    </repository>
    <repository name="sailfish_4.1.0.24_i486">
        <path project="sailfishos:4.1.0.24" repository="latest_i408"/>
        <arch>i586</arch>
    </repository>
    ```
````

   This will create build targets for the 4.1 and latest SaifishOS releases for
   all supported architectures.
3. Click "Save Changes", navigate back to the project overview, and you should see
   the build status for each target on the right hand side.
4. You can click on the package name, and then click on the build target status to
   see the current build log.
  
## Submitting your package to sailfishos:chum:testing

Once you are happy your builds are working, you can submit it for inclusion to chum:testing.

1. Navigate to the package, and click "Submit Package"
2. Enter the target project as `sailfishos:chum:testing`
3. Click Ok

> For packages submitted to chum, it is probably best to specify a particular git revision to be built.

Reviewer 3: ```suggestion
For packages submitted to chum, it is please specify a particular git revision to be built.
```
